internal/services/sql/sql_managed_instance_failover_group_resource.go

```diff
@@ -0,0 +1,404 @@
+		}
+
+		if err := d.Set("managed_instance_name", name); err != nil {
+			return fmt.Errorf("setting `managed_instance_name`: %+v", err)
+		}
+
+		if err := d.Set("read_write_endpoint_failover_policy", flattenSqlInstanceFailoverGroupReadWritePolicy(props.ReadWriteEndpoint)); err != nil {
+			return fmt.Errorf("setting `read_write_endpoint_failover_policy`: %+v", err)
+		}
+
+		if err := d.Set("readonly_endpoint_failover_policy", flattenSqlInstanceFailoverGroupReadOnlyPolicy(props.ReadOnlyEndpoint)); err != nil {
+			return fmt.Errorf("setting `read_only_endpoint_failover_policy`: %+v", err)
+		}
+
+		d.Set("role", string(props.ReplicationRole))
+
+		if err := d.Set("partner_managed_instances", flattenSqlInstanceFailoverGroupManagedInstancePairs(props.ManagedInstancePairs)); err != nil {
+			return fmt.Errorf("setting `partner_managed_instances`: %+v", err)
+		}
+
+		if err := d.Set("partner_regions", flattenSqlInstanceFailoverGroupPartnerRegions(props.PartnerRegions)); err != nil {
+			return fmt.Errorf("setting `partner_regions`: %+v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceSqlInstanceFailoverGroupDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sql.InstanceFailoverGroupsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.InstanceFailoverGroupID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, id.ResourceGroup, id.LocationName, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting %q: %+v", id, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for deletion of %q: %+v", id, err)
+	}
+
+	return err
+}
+
+func expandSqlInstanceFailoverGroupReadWritePolicy(d *pluginsdk.ResourceData) *sql.InstanceFailoverGroupReadWriteEndpoint {
+	vs := d.Get("read_write_endpoint_failover_policy").([]interface{})
+	v := vs[0].(map[string]interface{})
+
+	mode := sql.ReadWriteEndpointFailoverPolicy(v["mode"].(string))
+	graceMins := int32(v["grace_minutes"].(int))
+
+	policy := &sql.InstanceFailoverGroupReadWriteEndpoint{
+		FailoverPolicy: mode,
+	}
+
+	if mode != sql.ReadWriteEndpointFailoverPolicyManual {
+		policy.FailoverWithDataLossGracePeriodMinutes = utils.Int32(graceMins)
+	}
+
+	return policy
+}
+
+func flattenSqlInstanceFailoverGroupReadWritePolicy(input *sql.InstanceFailoverGroupReadWriteEndpoint) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	policy := make(map[string]interface{})
+
+	policy["mode"] = string(input.FailoverPolicy)
+
+	if input.FailoverWithDataLossGracePeriodMinutes != nil {
+		policy["grace_minutes"] = *input.FailoverWithDataLossGracePeriodMinutes
+	}
+	return []interface{}{policy}
+}
+
+func expandSqlInstanceFailoverGroupReadOnlyPolicy(d *pluginsdk.ResourceData) *sql.InstanceFailoverGroupReadOnlyEndpoint {
+	vs := d.Get("readonly_endpoint_failover_policy").([]interface{})
+	if len(vs) == 0 {
```

Comment:
We should check to see if `vs[0] == nil` too

Reply:
Thanks! 👍🏽